modules/claims_api/app/controllers/concerns/claims_api/v2/disability_compensation_validation.rb

```diff
@@ -833,8 +834,51 @@ def raise_exception_if_value_not_present(val, form_obj_description)
         )
       end
 
+      def validate_claim_process_type_bdd!
+        claim_process_type = form_attributes['claimProcessType']
+        if claim_process_type == 'BDD_PROGRAM'
+          date = form_attributes['claimDate'] || Time.find_zone!('Central Time (US & Canada)').today
+          claim_date = Date.parse(date.to_s)
```

Comment:
should we just wrap `form_attributes['claimDate']` in a Date.parse above? 

form_attributes is a string that we cast to a string to return a date and the `Time.find_zone` is a date we cast to a string that gets returned as a date in this line. we're duplicating the process somewhere, I feel like `Date.parse(form_attributes['claim_date'])` may make more sense?

```
(byebug) claim_date
Fri, 18 Dec 2020
(byebug) claim_date.class
Date
(byebug) Time.find_zone!('Central Time (US & Canada)').today
Fri, 20 Oct 2023
(byebug) Time.find_zone!('Central Time (US & Canada)').today.class
Date
```

whereas you could do
```
claim_date =  Date.parse(form_attributes['claimDate']) || Time.find_zone!('Central Time (US & Canada)').today

(byebug) claim_date
Fri, 18 Dec 2020
```

seems to function the same. with my example I'm seeing
```
{
    "errors": [
        {
            "title": "Unprocessable Entity",
            "detail": "Must have an activeDutyEndDate or anticipatedSeparationDate between 90 & 180 days from claim date."
        }
    ]
}
```

Reply:
I duplicated the logic for handling the claim date from validate_form_526_submission_claim_date! in line 105 of this file. I agree the existing approach could use some improvement...I'll review and make updates.

Reply:
The existing approach handles the case when claimDate is null, so leaving as-is for now.